### DISCUSSION TITLE
“photos.db” couldn’t be copied because you don’t have permission to access “osxphotos__7gk2b3z”

### DISCUSSION THREAD

**Discussion by tdavidesrifrance** [RhetTbull/osxphotos#1017] [Q&A]
Hello, I have been using osxphotos for years and I would like to begin by congratulating you on your work! 
I recently installed my mac (Ventura) on an external ssd to improve performance. Everything works fine.
Today, after installing osxphotos, my python script no longer works with the error below. I tried different things with permissions but nothing worked. Do you have an idea, please ?  Thanks in advance

Error copying/Volumes/Utilisateurs/lepouldu/Pictures/Photothèque.photoslibrary/database/photos.db to /var/folders/0y/4xkpjl2j02x_t3f4wlykjtxh0000gn/T/osxphotos__7gk2b3z/photos.db
Traceback (most recent call last):
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/osxphotos/photosdb/photosdb.py", line 618, in _copy_db_file
    FileUtil.copy(fname, dest_path)
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/osxphotos/fileutil.py", line 132, in copy
    raise OSError(error[1])
OSError: Error Domain=NSCocoaErrorDomain Code=513 "“photos.db” couldn’t be copied because you don’t have permission to access “osxphotos__7gk2b3z”." UserInfo={NSSourceFilePathErrorKey=/Volumes/Utilisateurs/lepouldu/Pictures/Photothèque.photoslibrary/database/photos.db, NSUserStringVariant=(
    Copy
), NSDestinationFilePath=/var/folders/0y/4xkpjl2j02x_t3f4wlykjtxh0000gn/T/osxphotos__7gk2b3z/photos.db, NSFilePath=/Volumes/Utilisateurs/lepouldu/Pictures/Photothèque.photoslibrary/database/photos.db, NSUnderlyingError=0x6000024bbed0 {Error Domain=NSPOSIXErrorDomain Code=1 "Operation not permitted"}}

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Volumes/Utilisateurs/lepouldu/Scripts/ExportPhotos/exportPhotosAll.py", line 71, in <module>
    photosdb = osxphotos.PhotosDB(library_path)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/osxphotos/photosdb/photosdb.py", line 324, in __init__
    self._tmp_db = self._copy_db_file(self._dbfile)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/osxphotos/photosdb/photosdb.py", line 626, in _copy_db_file
    raise Exception
Exception

**Comment by RhetTbull** [comment]
osxphotos needs to create some temporary files during execution and it looks like this process is failing. It's weird as osxphotos creates the temporary folder but then fails to copy the files.

Here are two things to try:

1. As your script is in `/Volumes/Utilisateurs/lepouldu/Scripts/ExportPhotos/`, you could create a temporary folder in here (`mkdir /Volumes/Utilisateurs/lepouldu/Scripts/ExportPhotos/tmp`) then add `--tmpdir /Volumes/Utilisateurs/lepouldu/Scripts/ExportPhotos/tmp` to your export options.
2. Ensure that your Terminal.app (or which ever terminal app you use) has "full disk permissions" granted in `System Settings > Privacy & Security > Full Disk Access`

If neither works, let me know and I'll explore some more to see if I can replicate this. I am running osxphotos on Ventura to export to external SSD and it works fine for me.

**Reply by RhetTbull** [reply]
@tdavidesrifrance are you still having problems with this?

**Reply by tdavidesrifrance** [reply]
hello @RhetTbull  Sorry for the delay. I found the solution. It was necessary to give "full disk permissions"  to Visual Studio Code in addition to Terminal. I also add permissions to Python and Python IDLE to be sure. Thanks for your help and your reactivity :)
